src/featureinfo.js

```diff
@@ -110,40 +109,38 @@ const Featureinfo = function Featureinfo(options = {}) {
     }
 
     return $(id).owlCarousel(carouselOptions);
-  }
+  };
 
   function getSelectionLayer() {
     return selectionLayer.getFeatureLayer();
   }
 
   function getSelection() {
     const selection = {};
-    if (selectionLayer.getFeatures()[0]) {
-      selection.geometryType = selectionLayer.getFeatures()[0].getGeometry().getType();
-      selection.coordinates = selectionLayer.getFeatures()[0].getGeometry().getCoordinates();
-      selection.id = selectionLayer.getFeatures()[0].getId();
-      selection.type = selectionLayer.getSourceLayer().get('type');
+    const firstFeature = selectionLayer.getFeatures()[0];
+    // console.log('first Feature: ', firstFeature);
+    if (firstFeature) {
+      selection.geometryType = firstFeature.getGeometry().getType();
+      // console.log('selection geometry type: ', selection.geometryType);
+      selection.coordinates = firstFeature.getGeometry().getCoordinates();
+      // console.log('selection coordinate: ', selection.coordinates);
+      selection.id = firstFeature.getId() || firstFeature.ol_uid;
```

Comment:
We need to allow 0 as a value:
selection.id = firstFeature.getId() != null ? firstFeature.getId() : firstFeature.ol_uid;

Reply:
@jokd sound good I will fix this.